### PR TITLE
Add ghc-8.10.7 support

### PR DESCRIPTION
**Pull request by jneira**
* 8.10.6 did not live much time 😄 
* I am getting error about boot packages locally, building with stack:

```
.....
In the dependencies for ghcide-1.4.1.0:
    directory needed, but this GHC boot package has been pruned (issue #4510); you need to add the
              package explicitly to extra-deps  (latest matching version is 1.3.6.2)
    ghc-boot needed, but this GHC boot package has been pruned (issue #4510); you need to add the
             package explicitly to extra-deps  (latest matching version is 9.0.1)
    process needed, but this GHC boot package has been pruned (issue #4510); you need to add the
            package explicitly to extra-deps  (latest matching version is 1.6.13.2)
    time needed, but this GHC boot package has been pruned (issue #4510); you need to add the
         package explicitly to extra-deps  (latest matching version is 1.12)
needed since ghcide is a build target.
...
Some different approaches to resolving this:

  * Recommended action: try adding the following to your extra-deps in D:\hls\stack-8.10.7.yaml:

- directory-1.3.6.2@sha256:6e5f3e0adfe94483d5754d97d741b7d23a5e085d47176075b1955d4fa78f37aa,2811
- ghc-boot-9.0.1@sha256:dc127d8b4b43eb54167f9befd51b2948e9084b8dfc18b88728148b719fa2287c,2347
- process-1.6.13.2@sha256:9733aa8a27b3e6c0f08a87a5c7287b3b73302d396f68f5655f9990a641368e3e,2845
- time-1.9.3@sha256:8f1b5448722a12a952248b356c9eb366e351226543d9086a2da71270522d5f45,5679

Plan construction failed.
```

### CODE REVIEW

**Comment by jneira** [comment]
it needs a choco update: https://github.com/Mistuke/GhcChoco/issues/16

File: ghcide/src/Development/IDE/Core/FileStore.hs [hunk 1]
```diff
@@ -289,7 +289,7 @@ typecheckParentsAction nfp = do
       Just rs -> do
         liftIO $ (log $ "Typechecking reverse dependencies for " ++ show nfp ++ ": " ++ show revs)
           `catch` \(e :: SomeException) -> log (show e)
-        () <$ uses GetModIface rs
```
Comment (pepeiborra) [inline ghcide/src/Development/IDE/Core/FileStore.hs:292]: we need to land this fix ASAP, as it's blocking everything

**Review by pepeiborra — APPROVED** [review]

**Comment by jneira** [comment]
stack does not support ghc-8
10.7 yet

**Comment by jneira** [comment]
stack downloads it but fails to install:

```
Preparing to download ghc-8.10.7 ...                                    ghc-8.10.7: download has begun                              ghc-8.10.7:   88.28 MiB / 205.11 MiB ( 43.04%) downloaded...                                                            ghc-8.10.7:  194.01 MiB / 205.11 MiB ( 94.59%) downloaded...                                                            ghc-8.10.7:  205.11 MiB / 205.11 MiB (100.00%) downloaded...                                                            Downloaded ghc-8.10.7.
Unpacking GHC into /root/.stack/programs/x86_64-linux/ghc-8.10.7.temp/ ...                                                                          Configuring GHC ...                   Installing GHC ...                  /root/.stack/programs/x86_64-linux/ghc-8.10.7/lib/ghc-8.10.7/bin/ghc-pkg: error while loading shared libraries: libtinfo.so.6: cannot open shared object file: No such file or directory
Installing GHC ...                  make[1]: *** [install_packages] Error 127
Installing GHC ...                  make: *** [install] Error 2
Installing GHC ...                  Received ExitFailure 2 when running

```

**Comment by jneira** [comment]
It is building now: https://app.circleci.com/pipelines/github/jneira/haskell-language-server/713/workflows/3a1af3e2-a491-4598-8a75-8daeaa7ef496/jobs/6165 :sweat_smile:

**Comment by jneira** [comment]
@MergifyIO backport 1.3.0-hackage

**Pull request merged**